content/tutorials/build-apps/build-a-desktop-wallet-in-js.md

```diff
@@ -0,0 +1,189 @@
+- Send [direct XRP payments](direct-xrp-payments.html), and provide feedback about the intended destination address, including:
+    - Displaying your account's available balance
+    - Verifying that the destination address is valid
+    - Validating the account has enough XRP to send
+    - Allowing you to specify a destination tag
+
+## Steps
+
+Before you begin, make sure you have the pre-requisites installed. Check your node version by running `node -v`. If you don't have node installed, you can download it [here](https://nodejs.org/en/download/).
+
+### Step 1: Setting up the project
+
+1. Navigate to the directory that you want to create the project in.
+2. Run 
+   ```sh
+    yarn create vite-app <project-name>
+    ```
+    or
+    ```sh
+    npm init vite-app <project-name>
+    ```
+    to create a new project.
+
+3. Navigate to your project directory and copy this [package.json](_code-samples/build-a-wallet/js/package.json) file to your project.
+        - Alternatively you can also do `yarn add <package-name>` or `npm install <package-name>` for each individual package to add them to your `package.json` file.
+  
+4. Run `yarn` or `npm i` to install the packages.
+
+5. Create a new file `.env` in the root directory of the project and add the following variables:
+
+```bash
+CLIENT="wss://s.altnet.rippletest.net/" // or any other server
+EXPLORER_NETWORK="testnet" // or "mainnet"
+SEED="sEd7B2Jm7RBhhh5i88SH38aZB8ot8Kq" // Replace with your seed
+```
+6. Change the seed to your own seed. You can get a testnet account from [here](https://xrpl.org/xrp-test-net-faucet.html).
+
+
+### Step 2: Running the App
+
+In this step, we will set up the vite bundler to run the application. We've used Vite to bundle the app, if you're unfamiliar with Vite, you can check out the [Vite documentation](https://vitejs.dev/guide/) to learn more.
+
+1. To set up vite bundler, copy the code written below to the vite.config.js file. xrpl.js requires some configuration to run with the vite bundler. You can find the configuration in the code below or you can check out the [xrpl.js documentation](https://github.com/XRPLF/xrpl.js/blob/main/UNIQUE_SETUPS.md#using-xrpljs-with-vite-react).
+
+{{ include_code("_code-samples/build-a-wallet/js/vite.config.js", language="js") }}
+
+2. Add script to package.json
+
+In your package.json file, add:
+
+```json
+"scripts": {
+    "dev": "vite"
+}
+```
+
+Now the basic setup of your application is ready, after completing the next steps of this tutorial you can run this command
+
+```bash
+yarn dev ## or npm run dev
+```
+
+in your terminal which will start your app in development mode. Your terminal will display an URL using which you can verify your changes in real-time.
+
+If you get stuck building while doing this tutorial, or working on another project, feel free to ask for help in the XRPL's [Developer Discord](https://discord.com/invite/KTNmhJDXqa).
```

Comment:
If this works right away, I'd keep this here, otherwise I'd move this comment about using yarn dev or npm run dev until you know it'll work. (e.g. after completing the 1st set of steps)

I'd also separate the `yarn dev` and `npm run dev` into two separate lines in order to make it harder to miss the npm run dev instruction, e.g.
`yarn dev`
or
`npm run dev`